### PR TITLE
[WIP] fix: ast compare

### LINKED ISSUE #514: AST_COMPARE Troubleshooting
A few days back, I inherited an older code base that I really wanted to make prettier before diving in! I found this plugin and thought I'd give it a try (despite the alpha status), but unfortunately there's a few parts of the code base that are tripping it up. I dove in to the code to see if I could figure out what was going on, but I hit a snag early on with the tests not seeming to run correctly!

Steps To Reproduce:

```bash
git clone https://github.com/prettier/plugin-php.git
cd plugin-php
yarn
yarn test
AST_COMPARE=1 yarn test
```

With and without the `AST_COMPARE=1` setting, the result is the same: all tests pass. I don't think this is correct, even for some of the existing test cases!

Long story short, some logging showed that the comparison wasn't even running, so I made some changes, got it working for me, and pushed that code here:

https://github.com/ionrocks/plugin-php/tree/testing-updates

I didn't make a pull request because I don't know this code base very well, and am hoping someone will tell me if I'm missing something obvious. 😃

### PR DESCRIPTION
**Pull request by evilebottnawi**
fixes #514

### CODE REVIEW

File: tests/comments/__snapshots__/jsfmt.spec.js.snap [hunk 1]
```diff
@@ -1200,9 +1200,9 @@ include // Comment
 
 // Comment
 include // Comment
-    // Comment
-    "test.php"; // Comment
-    // Comment // Comment
+// Comment
+"test.php"; // Comment
+// Comment // Comment
```
[the remark this answers is not in the document]
Comment (czosel) [reply tests/comments/__snapshots__/jsfmt.spec.js.snap]: Ok!

File: tests/comments/__snapshots__/jsfmt.spec.js.snap [hunk 2]
```diff
@@ -1226,8 +1226,7 @@ exports[`halt.php 1`] = `
 /*Comment*/__halt_compiler/*Comment*/(/*Comment*/)/*Comment*/;~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 <?php
 
-/*Comment*/ __halt_compiler(); /*Comment*/ /*Comment*/ /*Comment*/
-
+/*Comment*/ /*Comment*/ /*Comment*/ /*Comment*/ __halt_compiler();
```
Comment (evilebottnawi) [inline tests/comments/__snapshots__/jsfmt.spec.js.snap]: Looks some ugly, but it is edge and let's fix it in other PR

File: tests/comments/__snapshots__/jsfmt.spec.js.snap (hunk 1 above)
[the remark this answers is not in the document]
Comment (evilebottnawi) [reply tests/comments/__snapshots__/jsfmt.spec.js.snap]: Issue https://github.com/prettier/plugin-php/issues/629

**Comment by evilebottnawi** [comment]
We fix it :star:

**Comment by evilebottnawi** [comment]
All problems with parser in todo in code

**Pull request merged**